tests/moe_ep/test_weight_pack_union.py

```
@@ -0,0 +1,124 @@
+    assert isinstance(pre, PrequantizedMoEWeights)
+
+
+def test_nvfp4_preprocess_dispatches_prequantized_shapes():
+    # The prequantized branch validates scale shapes before any GPU work:
+    # a wrong scale shape must raise from the prequantized path (proving the
+    # isinstance dispatch selected it), not fall through to re-quantization.
+    from flashinfer.moe_ep.backends.mega.kernel.nvfp4_cutedsl.weights import (
+        preprocess_mega_weights,
+    )
+
+    w13, w2, s13, s2 = _packed_tensors()
```

Ruff (0.15.21)

[warning] Unpacked variable `s13` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)